ksql-engine/src/main/java/io/confluent/ksql/codegen/CodeGenRunner.java

```diff
@@ -188,7 +182,9 @@ protected Object visitDereferenceExpression(DereferenceExpression node, Object c
         throw new RuntimeException(
             "Cannot find the select field in the available fields: " + node.toString());
       }
-      updateFunctionArgTypesAndParams(schemaField.get());
+      parameters.add(new ParameterType(
```

Comment:
There are multiple copies of this same code - maybe it should be a method?

e.g.
```
private void addParameter(final Field field) {
      parameters.add(new ParameterType(
          SchemaUtil.getJavaType(field.schema()),
          field.name().replace(".", "_")));
    }
```